src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -767,5 +811,16 @@ def disk_configurations(vm_type, existing_disk_cids=[])
       )
       disk_configurations.push(ephemeral_disk_config)
     end
+
+    #This subnet will be used in create_logical_router_port.
+    # ip_addresses is going to be a gateway IP for subnet - aka IP for router.
+    def create_subnet_obj(range, gateway)
+      if (!range.nil? && range.include?('/'))
+        _, mask = range.split("/")
+        return NSXT::IPSubnet.new({:ip_addresses => [gateway],
+                                   :prefix_length => mask.to_i})
```

Comment:
Ruby has multiple modes of typecasting that are well explained here:

https://idiosyncratic-ruby.com/54-try-converting.html

In short though `#to_i` is almost never what you want to use with user input since:

> This method never raises an exception when base is valid

Instead you probably want something like `Integer(mask, 10)`.